quickwit/quickwit-storage/src/cache/byte_range_cache.rs

```diff
@@ -106,13 +109,13 @@ impl<T: 'static + ToOwned + ?Sized + Ord> NeedMutByteRangeCache<T> {
             return;
         }
 
+        // try to find a block with which we overlap (and not just touch)
         let start_key = CacheKey::from_borrowed(tag.borrow(), byte_range.start);
-        let end_key = CacheKey::from_borrowed(tag.borrow(), byte_range.end);
-
         let first_matching_block = self
-            .get_block(&start_key, byte_range.start)
+            .get_block(&start_key, byte_range.start + 1)
             .map(|(k, _v)| k);
 
+        let end_key = CacheKey::from_borrowed(tag.borrow(), byte_range.end - 1);
```

Comment:
my understanding is that with this, we remove (as opposed to merge) any data that is overlapping (not just is included in) with the slice we are trying to put. 

Is my understanding correct?
If so can we comment this more clearly?

possibly isolate in a different method:
`fn remove_overlapping_slices(&mut self, ....)`

Reply:
yes: correction.
we merge any data that is overlapping (not just touches) with the slice we are trying to put.

could we comment what we are doing, or even better isolate it into a method?